components/scream/src/scream_config.h.in

```diff
@@ -10,36 +10,18 @@
 // If defined, enable floating point exceptions.
 #cmakedefine SCREAM_FPE
 
-// If defined, scream::pack objects check indices against bounds.
```

Comment:
Yeah, you're right. I just noticed these cmakedefines randomly, and figured I purged them, since they're never used. I think they were moved to EKAT when we extract it as a submodule long time ago. If you prefer, I can revert this commit, and leave the config.h cleanup for another time.

Reply:
Ok, as long as these are now available through EKAT, I'm fine with it.

Reply:
Well, hold on. Some are, but some I should probably bring them back, or clean them from our CMakeLists.txt too. In particular:
```
#cmakedefine SCREAM_MPI_ERRORS_ARE_FATAL
#cmakedefine SCREAM_CONSTEXPR_ASSERT
#cmakedefine SCREAM_MIMIC_GPU
```
are available as `EKAT_<XYZ>`.

`SCREAM_STRICT_FP` was not even a CMake var. Ekat makes `<PREFIX>_TEST_STRICT_FP` available (and we build Ekat with PREFIX=SCREAM), but we never use it.

The remaining two are declared in our CMakeLists.txt, but never used anywhere. I should either purge the CMake vars as well, or bring back the cmakedefines for them. What do you think?

Reply:
It seems like `SCREAM_PACK_CHECK_BOUNDS` should be in EKAT since that's where the Pack implementation lives. I'm ok to remove the scream one.

I remember adding SCREAM_POSSIBLY_NO_PACK_SIZE way back when I was doing the linear interp performance testing. We discovered that pack_size=1 was better on SKX for lin interp but not on other architectures. I think this concept is potentially useful to have when we have algorithms that don't sometimes don't perform well with the default pack size, so I'd be in favor of keeping it even though it is currently unused and a bit confusing.

Reply:
Ok, I'll bring the possibly-no-pack one back.